spring-data-neo4j/src/test/java/org/springframework/data/neo4j/examples/restaurants/RestaurantIT.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c)  [2011-2016] "Pivotal Software, Inc." / "Neo Technology" / "Graph Aware Ltd."
+ *
+ * This product is licensed to you under the Apache License, Version 2.0 (the "License").
+ * You may not use this product except in compliance with the License.
+ *
+ * This product may include a number of subcomponents with
+ * separate copyright notices and license terms. Your use of the source
+ * code for these subcomponents is subject to the terms and
+ * conditions of the subcomponent's license, as noted in the LICENSE file.
+ *
+ */
+
+
+package org.springframework.data.neo4j.examples.restaurants;
+
+import static org.apache.webbeans.util.Asserts.assertNotNull;
+import static org.junit.Assert.*;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import org.junit.After;
+import org.junit.Assume;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.neo4j.ogm.service.Components;
+import org.neo4j.ogm.testutil.MultiDriverTestClass;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.geo.Distance;
+import org.springframework.data.geo.Metrics;
+import org.springframework.data.geo.Point;
+import org.springframework.data.neo4j.examples.restaurants.context.RestaurantContext;
+import org.springframework.data.neo4j.examples.restaurants.domain.Restaurant;
+import org.springframework.data.neo4j.examples.restaurants.repo.RestaurantRepository;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+@ContextConfiguration(classes = {RestaurantContext.class})
+@RunWith(SpringJUnit4ClassRunner.class)
+@DirtiesContext
+
+/**
+ * @author Jasper Blues
+ */
+public class RestaurantIT extends MultiDriverTestClass {
+
+	@Autowired
+	private RestaurantRepository restaurantRepository;
+
+	@After
+	public void tearDown()
+	{
+		restaurantRepository.deleteAll();
+	}
+
+
+	/**
+	 * @see DATAGRAPH-561
+	 */
+	@Test
+	public void shouldFindRestaurantsNear_nameParameterFirst() {
+		Assume.assumeTrue(Components.neo4jVersion() >= 3);
```

Comment:
With SDN tests you generally need to mark multi step interactions like this (repo.save then repo.find) inside a transaction. Otherwise it will create two sessions each with their own transaction. Of course if that's the intended usage then that's fine. Just depends!